### DISCUSSION TITLE
Using --modtime to keep multiple commands in sync

### DISCUSSION THREAD

**Discussion by dimo414** [dimo414/bkt#26] [Ideas]
`--scope` can be used to group invocations together and isolate them from other similar calls, but it's inconvenient to use it as a mechanism to flush the cache (#5) because all commands have to coordinate using the same scope name. Something like this works, but comes with extra file I/O and subshells:

```shell
scope_file=$(mktemp)
echo my-script-1 > "$scope_file"

bkt --scope="$(< "$scope_file")" --ttl=1h -- foo
bkt --scope="$(< "$scope_file")" --ttl=1h -- bar

do_something --that-invalidates --the-cache
echo my-script-2 > "$scope_file"

bkt --scope="$(< "$scope_file")" --ttl=1h -- foo
bkt --scope="$(< "$scope_file")" --ttl=1h -- bar
```

With `--modtime` added in 0.6.0 this strategy gets a little simpler and more efficient:

```shell
watch_file=$(mktemp)

bkt --modtime="$watch_file" --ttl=1h -- foo
bkt --modtime="$watch_file" --ttl=1h -- bar

do_something --that-invalidates --the-cache
touch "$watch_file"

bkt --modtime="$watch_file" --ttl=1h -- foo
bkt --modtime="$watch_file" --ttl=1h -- bar
```

Please share any examples you have of applying this pattern!

**Comment by jankatins** [comment]
Nowadays you can also `export BKT_SCOPE=my-script-1` (and update it inbetween) which removes the need for a scope_file in the first example (the second example would also work in case you have to coordinate across multiple processes and not only a single shell script).

**Reply by dimo414** [reply]
That's true, if all the processes share the same parent environment you can update the `BKT_SCOPE` environment variable to similar effect, but the benefit of using a file's `--modtime` is you don't have that constraint.
